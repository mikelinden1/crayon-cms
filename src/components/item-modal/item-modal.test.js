import ItemModal from './item-modal';
import sinon from 'sinon';

const props = {
    open: true,
    newItem: {},
    config: {
        itemProps: [
            { name: 'name', type: 'text' },
            { name: 'dropdown', type: 'dropdown', options: [{ value: 'val1', label: 'Val1' }, { value: 'val2', label: 'Val2' }] }
        ],
        itemName: 'Test'
    },
    currentModule: 'widgets',
    actions: {
        saveNewItem: () => {},
        saveEditItem: () => {},
        closeItemModal: () => {}
    }
};

it('ItemModal matches snapshot', () => {
    const wrapper = shallow(<ItemModal {...props} />);
    expect(wrapper).toMatchSnapshot();
});

// it('ItemModal closes the modal on cancel', () => {
//     const spy = sinon.spy();
//     props.actions.closeItemModal = spy;

//     const store = mockStore({});

//     const wrapper = mountWrap(
//         <Provider store={store}>
//             <ItemModal {...props} />
//         </Provider>
//     );

//     wrapper.find('.cancel-btn').simulate('click');

//     expect(spy.calledOnce).toBeTruthy();
// });