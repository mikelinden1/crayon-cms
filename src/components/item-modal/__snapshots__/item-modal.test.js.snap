// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemModal matches snapshot 1`] = `
<Modal
  autoFocus={true}
  backdrop={true}
  backdropTransition={
    Object {
      "mountOnEnter": true,
      "timeout": 150,
    }
  }
  centered={false}
  fade={true}
  isOpen={true}
  keyboard={true}
  modalTransition={
    Object {
      "timeout": 300,
    }
  }
  onClosed={[Function]}
  onOpened={[Function]}
  role="dialog"
  toggle={[Function]}
  zIndex={1050}
>
  <form
    onSubmit={[Function]}
  >
    <ModalHeader
      closeAriaLabel="Close"
      tag="h5"
      toggle={[Function]}
      wrapTag="div"
    >
      New
       
      Test
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      <Connect(FieldGroup)
        key="name-field"
        name="name"
        type="text"
      />
      <Connect(FieldGroup)
        key="dropdown-field"
        name="dropdown"
        options={
          Array [
            Object {
              "label": "Val1",
              "value": "val1",
            },
            Object {
              "label": "Val2",
              "value": "val2",
            },
          ]
        }
        type="dropdown"
      />
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        className="save-btn"
        color="primary"
        tag="button"
        type="submit"
      >
        Save
      </Button>
       
       
      <Button
        className="cancel-btn"
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Cancel
      </Button>
    </ModalFooter>
  </form>
</Modal>
`;
